Sources/Exporters/OpenTelemetryProtocol/logs/OtlpHttpLogExporter.swift

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+// 
+
+import Foundation
+import Logging
+import NIO
+import NIOHPACK
+import OpenTelemetryApi
+import OpenTelemetrySdk
+
+public func defaultOltpHttpLoggingEndpoint() -> URL {
+    URL(string: "http://localhost:4318/v1/logs")!
+}
+
+public class OtlpHttpLogExporter : LogRecordExporter {
+    let endpoint: URL
+    private let httpClient: HTTPClient
+    var pendingLogRecords: [ReadableLogRecord] = []
+    
+    public init(endpoint: URL = defaultOltpHttpLoggingEndpoint(), useSession: URLSession? = nil) {
+        self.endpoint = endpoint
+        if let providedSession = useSession {
+            self.httpClient = HTTPClient(session: providedSession)
+        } else {
+            self.httpClient = HTTPClient()
+        }
+    }
+    
+    public func export(logRecords: [OpenTelemetrySdk.ReadableLogRecord]) -> OpenTelemetrySdk.ExportResult {
+        pendingLogRecords.append(contentsOf: logRecords)
+        return self.flush()
+    }
+    
+    public func shutdown() {
+    }
+    
+    public func forceFlush() -> OpenTelemetrySdk.ExportResult {
+        self.flush()
+    }
+    
+    public func flush() -> ExportResult {
+        var exporterResult: ExportResult = .failure
+        let logRecords = pendingLogRecords
+        pendingLogRecords = []
+            
+        let body = Opentelemetry_Proto_Collector_Logs_V1_ExportLogsServiceRequest.with { request in
+            request.resourceLogs = LogRecordAdapter.toProtoResourceRecordLog(logRecordList: logRecords)
+        }
+
+        var request = URLRequest(url: endpoint)
+        request.httpMethod = "POST"
+        request.httpBody = try? body.serializedData()
```

Comment:
It may be best to wrap this in a do/catch and print out the error. Sending nil data up isn't going to help anybody.

Reply:
True. can you change it @kevinearls?

Reply:
Sure, I will work on as many of these as I can this afternoon.